test/infrastructure/docker/docker/machine.go

```diff
@@ -124,10 +128,22 @@ func (m *Machine) Create(role string, version *string) error {
 				nil,
 				nil,
 			)
-			return errors.WithStack(err)
+			if err != nil {
+				return errors.WithStack(err)
+			}
 		default:
 			return errors.Errorf("unable to create machine for role %s", role)
 		}
+		// After creating a node we need to wait a small amount of time until crictl does not return an error.
+		// This fixes an issue where we try to kubeadm init too quickly after creating the container.
+		err = wait.PollImmediate(500*time.Millisecond, 2*time.Second, func() (bool, error) {
```

Comment:
my understanding is that it depends on the host machine spec so one's mileage may vary.
i guess if the same flakes are seen this can be bumped, but if i remember correctly `crictl ps` implements retries too.

/lgtm

Reply:
/hold

i want to check on crictl ps implementing a retry. I'd much rather rely on that retry than my own home grown one!

Reply:
Would it be better to handle this in the cloud-init conversion code? For regular infra providers we are now relying on the ordering of the systemd units associated with cloud-init phases to ensure that containerd is up prior to executing the script that runs kubeadm.

Reply:
interesting, this could also be implemented as a pre-command to kubeadm init as well and then we wouldn't need any code changes at all.

Reply:
Because this is not using the image-builder, and because kind nodes have a different way of configuring CNI and because we don't really have cloud-init, that this makes the most amount of sense for this provider. This code is located directly after the `docker run` command is issued. I think there is an argument for refactoring into a "node ready" function that runs a series of commands to ensure the container is ready, but I think shoe-horning this code into what the image-builder is doing when we aren't using the image builder will cause some confusion down the road.